src/deluge/gui/menu_item/generate/dmenus/oscillator/sample.py

```diff
@@ -0,0 +1,87 @@
+from dmui.dsl import *
+
+_available_txt = 'Oscillator has its type set to <string-for name="STRING_FOR_SAMPLE">SAMPLE</string-for>'
+
+recorder = Menu(
+    "osc::AudioRecorder",
+    "audioRecorderMenu",
+    ["{name}"],
+    "STRING_FOR_RECORD_AUDIO",
+    "oscillator/sample/recorder.md",
+)
+
+reverse = Menu(
+    "sample::Reverse",
+    "sampleReverseMenu",
+    ["{name}", "{title}"],
+    "STRING_FOR_SAMP_REVERSE_MENU_TITLE",
+    "oscillator/sample/reverse.md",
+    name="STRING_FOR_REVERSE",
+    available_when=_available_txt,
+)
+
+repeat = Menu(
+    "sample::Repeat",
+    "sampleRepeatMenu",
+    ["{name}", "{title}"],
+    "STRING_FOR_SAMP_REPEAT_MENU_TITLE",
+    "oscillator/sample/repeat.md",
+    name="STRING_FOR_REPEAT_MODE",
+    available_when=_available_txt,
+)
+
+start = Menu(
+    "sample::Start",
+    "sampleStartMenu",
+    ["{name}"],
+    "STRING_FOR_START_POINT",
+    "oscillator/sample/start_point.md",
+    available_when=_available_txt,
+)
+
+end = Menu(
+    "sample::End",
+    "sampleEndMenu",
+    ["{name}"],
+    "STRING_FOR_END_POINT",
+    "oscillator/sample/end_point.md",
+    available_when=_available_txt,
+)
+
+transpose = Menu(
+    "sample::Transpose",
+    "sourceTransposeMenu",
+    ["{name}", "{title}", "params::LOCAL_OSC_A_PITCH_ADJUST"],
+    "STRING_FOR_OSC_TRANSPOSE_MENU_TITLE",
+    "oscillator/sample/transpose.md",
+    name="STRING_FOR_TRANSPOSE",
+)
+
+pitch_speed = Menu(
+    "sample::PitchSpeed",
+    "samplePitchSpeedMenu",
+    ["{name}"],
```

Comment:
What's happening in this one, why isn't there a title?

Reply:
I don't know, that's just how the C++ constructor is

Reply:
Ah ok, thought it was shenanigans with the dsl